x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/core/expression/MapExpression.java

```diff
@@ -147,4 +147,21 @@ public String toString() {
         String str = entryExpressions.stream().map(String::valueOf).collect(Collectors.joining(", "));
         return "{ " + str + " }";
     }
+
+    public Map<String, Object> toFoldedMap(FoldContext ctx) throws IllegalStateException {
```

Comment:
I can't make this `foldable` unfortunately, otherwise the planner will try to fold function parameters, and too many things will break.